### PR TITLE
Add Hashing Algorithm to Image Model

### PR DESCRIPTION
**Pull request by zackt25**
Adding in hash algorithm field for #144

### CODE REVIEW

**Comment by zackt25** [comment]
I am going to rebase once the other PRs get merged so I have good migration files

File: nautobot_device_lifecycle_mgmt/forms.py [hunk 1]
```diff
@@ -369,6 +369,10 @@ class SoftwareImageLCMFilterForm(BootstrapMixin, forms.ModelForm):
         required=False,
     )
     default_image = forms.BooleanField(required=False, widget=StaticSelect2(choices=BOOLEAN_WITH_BLANK_CHOICES))
+    hash_algorithm = forms.CharField(
+        required=False,
+        label="Hash Algorithm",
```
Comment (progala) [inline nautobot_device_lifecycle_mgmt/forms.py:374]:
```suggestion
        label="Hashing Algorithm",
```

File: nautobot_device_lifecycle_mgmt/migrations/0010_softwareimagelcm_hash_algorithm.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    dependencies = [
+        ("nautobot_device_lifecycle_mgmt", "0009_software_remove_image_fields"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="softwareimagelcm",
+            name="hash_algorithm",
+            field=models.CharField(blank=True, default="", max_length=15),
```
Comment (progala) [inline nautobot_device_lifecycle_mgmt/migrations/0010_softwareimagelcm_hash_algorithm.py:16]:
```suggestion
            field=models.CharField(blank=True, default="", max_length=32),
```

File: nautobot_device_lifecycle_mgmt/models.py [hunk 3]
```diff
@@ -276,6 +276,7 @@ class SoftwareImageLCM(PrimaryModel):
     object_tags = models.ManyToManyField(to="extras.Tag", related_name="+", blank=True)
     download_url = models.URLField(blank=True, verbose_name="Download URL")
     image_file_checksum = models.CharField(blank=True, max_length=256, verbose_name="Image File Checksum")
+    hash_algorithm = models.CharField(default="", blank=True, max_length=15, verbose_name="Hash Algorithm")
```
Comment (progala) [inline nautobot_device_lifecycle_mgmt/models.py:279]:
```suggestion
    hash_algorithm = models.CharField(default="", blank=True, max_length=32, verbose_name="Hashing Algorithm")
```

File: nautobot_device_lifecycle_mgmt/tables.py [hunk 4]
```diff
@@ -167,6 +167,13 @@ class SoftwareImageLCMTable(BaseTable):
         viewname="extras:tag_list", url_params={"slug": ("object_tags", "slug")}, verbose_name="Object Tags"
     )
     default_image = BooleanColumn()
+    download_url = tables.TemplateColumn(
+        template_code="""{% if record.download_url %}
+                    <a href="{{ record.download_url }}" target="_blank" data-toggle="tooltip" data-placement="left" title="{{ record.download_url }}">
+                        <span class="mdi mdi-open-in-new"></span>
+                    </a>{% else %} — {% endif %}""",
+        verbose_name="Download URL",
+    )
```
Comment (progala) [inline nautobot_device_lifecycle_mgmt/tables.py:176]: This should come in a separate PR.

File: nautobot_device_lifecycle_mgmt/tables.py [hunk 5]
```diff
@@ -180,6 +187,7 @@ class Meta(BaseTable.Meta):  # pylint: disable=too-few-public-methods
             "device_type_count",
             "object_tag_count",
             "default_image",
+            "download_url",
```
Comment (progala) [inline nautobot_device_lifecycle_mgmt/tables.py:190]: This should come in a separate PR.

File: nautobot_device_lifecycle_mgmt/templates/nautobot_device_lifecycle_mgmt/softwareimagelcm.html [hunk 6]
```diff
@@ -49,6 +49,16 @@ <h2>{% block title %}{{ object }}{% endblock %}</h2>
                 {% endif %}
                 </td>
             </tr>
+            <tr>
+                <td>Hash Algorithm</td>
```
Comment (progala) [inline nautobot_device_lifecycle_mgmt/templates/nautobot_device_lifecycle_mgmt/softwareimagelcm.html:53]:
```suggestion
                <td>Hashing Algorithm</td>
```

**Review by progala — COMMENTED** [review]
I see we generally say "Hash function" but "Hashing algorithm", so I made some suggestions based around that.

I think 32 chars for the new field might be a bit safer.

Code related to the URL field fix should be included in a separate PR.

File: nautobot_device_lifecycle_mgmt/tables.py (hunk 4 above)
Comment (zackt25) [reply nautobot_device_lifecycle_mgmt/tables.py]: I've removed and will create new PR

File: nautobot_device_lifecycle_mgmt/tables.py (hunk 5 above)
Comment (zackt25) [reply nautobot_device_lifecycle_mgmt/tables.py]: I've removed and will create new PR

File: nautobot_device_lifecycle_mgmt/forms.py (hunk 1 above)
Comment (zackt25) [reply nautobot_device_lifecycle_mgmt/forms.py]: Fixed

File: nautobot_device_lifecycle_mgmt/migrations/0010_softwareimagelcm_hash_algorithm.py (hunk 2 above)
Comment (zackt25) [reply nautobot_device_lifecycle_mgmt/migrations/0010_softwareimagelcm_hash_algorithm.py]: Fixed

File: nautobot_device_lifecycle_mgmt/models.py (hunk 3 above)
Comment (zackt25) [reply nautobot_device_lifecycle_mgmt/models.py]: Fixed

**Review by progala — APPROVED** [review]
LGTM!

**Pull request merged**